### PR TITLE
Integration tests for `llms` crate, with basic Anthropic test.

### PR DESCRIPTION
**Pull request by Jeadie**
## 🗣 Description
 - Add integration tests to `llms` crate to test core model functionality.
 - Add Anthropic as first model.
 - Found from testing, Anthropic was still explicitly setting a public model name in the response, separate from the anthropic model id. This is no longer needed since https://github.com/spiceai/spiceai/pull/3458. It is removed in this PR. 
 
NOTE:  Need `SPICE_SECRET_ANTHROPIC_API_KEY` added to Github secrets.

### CODE REVIEW

File: crates/llms/tests/llms/anthropic.rs [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+use llms::{
+    anthropic::{Anthropic, AnthropicConfig},
+    chat::Chat,
+};
+
+pub(crate) fn create_chat(model_id: Option<&str>) -> Result<Arc<dyn Chat>, OpenAIError> {
+    let cfg = AnthropicConfig::default()
+        .with_api_key(std::env::var("SPICE_ANTHROPIC_API_KEY").ok())
+        .with_auth_token(std::env::var("SPICE_ANTHROPIC_AUTH_TOKEN").ok());
+    println!("AnthropicConfig: {cfg:?}");
```
Comment (sgrebnov) [inline crates/llms/tests/llms/anthropic.rs]:
```suggestion
```

Or `tracing`

File: crates/llms/tests/llms/mod.rs [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+/// Run a single [`TestCase`] for a model.
+#[allow(clippy::expect_used, clippy::expect_fun_call)]
+async fn run_test_case(
+    test: &TestCase,
+    model_name: &'static str,
+    model: Arc<dyn Chat>,
+) -> Result<(), anyhow::Error> {
+    let test_name = test.name;
+    println!("Running test {test_name}/{model_name} with {:?}", test.req);
```
Comment (sgrebnov) [inline crates/llms/tests/llms/mod.rs]: Use tracing

File: crates/llms/tests/llms/mod.rs [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+        test_case!("basic", json!({
+            "model": "not_needed",
+            "messages": [
+                {
+                    "role": "user",
+                    "content": "Say Hi"
+                }
+            ]
+        }), vec![
+            ("message_keys", "$.choices[*].message['role', 'tool_calls', 'refusal', 'function_calls']"),
```
Comment (sgrebnov) [inline crates/llms/tests/llms/mod.rs]: Nice 👍

File: crates/llms/tests/llms/mod.rs [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+            format!("{test_name}_{model_name}_{id}"),
+            serde_json::to_string_pretty(&resp_ptr).expect("Failed to serialize snapshot")
+        );
+    }
+    Ok(())
+}
+
+#[tokio::test]
+#[allow(clippy::expect_used, clippy::expect_fun_call)]
+async fn run_all_tests() {
```
Comment (sgrebnov) [inline crates/llms/tests/llms/mod.rs:197]: Should we allow running tests only for specific model - Anthropic, OpenAI, HF? Can we have similar approach but single test per model provider or some configuration how to control this?

File: crates/llms/tests/llms/mod.rs [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+        TestCase {
+            name: $name,
+            req: serde_json::from_value($req)
+                .expect(&format!("Failed to parse request in test case '{}'", $name)),
+            json_path: $jsonpaths,
+        }
+    };
+}
+
+lazy_static! {
```
Comment (phillipleblanc) [inline crates/llms/tests/llms/mod.rs]: Use `LazyLock` instead: https://doc.rust-lang.org/std/sync/struct.LazyLock.html

File: crates/runtime/Cargo.toml [hunk 6]
```diff
@@ -134,11 +134,12 @@ azure_storage_blobs = "0.21.0"
 bollard = "0.18.0"
 flightrepl = { path = "../flightrepl" }
 insta.workspace = true
+jsonpath = "0.1.1"
```
Comment (phillipleblanc) [inline crates/runtime/Cargo.toml]: Why is this required as a dependency of runtime?

**Review by phillipleblanc — DISMISSED** [review]

File: crates/runtime/Cargo.toml (hunk 6 above)
Comment (Jeadie) [reply crates/runtime/Cargo.toml]: Bad merge conflict resolution when this branch was based atop https://github.com/spiceai/spiceai/pull/3576. Will remove.

Comment (Jeadie) [reply crates/runtime/Cargo.toml]: It was the package used before `jsonpath_rust`

File: crates/llms/tests/llms/mod.rs (hunk 4 above)
Comment (Jeadie) [reply crates/llms/tests/llms/mod.rs:197]: I plan to do this in a separate PR where I do two things:
1. Add additional providers
2. Solve this exact problem.

**Review by sgrebnov — DISMISSED** [review]

**Review by sgrebnov — APPROVED** [review]
